disnake/member.py

```diff
@@ -619,6 +621,17 @@ def guild_avatar(self) -> Optional[Asset]:
             return None
         return Asset._from_guild_avatar(self._state, self.guild.id, self.id, self._avatar)
 
+    @property
+    def guild_banner(self) -> Optional[Asset]:
```

Comment:
a `display_banner` field would be a good QOL addition, which also keeps it in line with avatars and decorations

Reply:
This has no reason to exist, as it is non-functional, as the API only returns the user banners on fetched users.

Reply:
The gateway doesn't send user banners, but REST does, so this works fine with `guild.fetch_member` which also returns user data for the requested member.
The caveat here is that the user can't be cached yet, since `store_user` will prefer the cached user (without banner) over the passed user data (with banner):
https://github.com/DisnakeDev/disnake/blob/93de3c8726a363095a881fb8f728fc4cd55c7494/disnake/member.py#L328-L330
It's out of scope here, but `display_banner` isn't completely useless either way.

Reply:
I don't like how it functions, given its a bit of a footgun in its current state.

Reply:
> given its a bit of a footgun in its current state

hmm, fair enough; it is somewhat inconsistent.
In that case, I suppose `display_banner` would be better suited as a `TODO` for the future, once the `MemberWithUserPayload` handling improves a bit. Bit unfortunate, but you're right.